cli/flox/src/commands/upload.rs

```diff
@@ -0,0 +1,134 @@
+use std::path::PathBuf;
+
+use anyhow::{bail, Result};
+use bpaf::Bpaf;
+use flox_rust_sdk::flox::Flox;
+use flox_rust_sdk::providers::publish::{BinaryCache, NixCopyCache};
+use tracing::instrument;
+use url::Url;
+
+use crate::config::Config;
+use crate::subcommand_metric;
+use crate::utils::dialog::{Dialog, Spinner};
+use crate::utils::message;
+
+#[derive(Bpaf, Clone)]
+pub struct Upload {
+    #[bpaf(external(cache_args))]
+    cache: CacheArgs,
+
+    #[bpaf(external(upload_store_path))]
+    store_path: UploadStorePath,
+}
+
+#[derive(Debug, Bpaf, Clone)]
+struct CacheArgs {
+    #[bpaf(long("cache"))]
+    url: Url,
+
+    #[bpaf(long("signing-key"))]
+    key_file: PathBuf,
+}
+
+#[derive(Debug, Bpaf, Clone)]
+struct UploadStorePath {
+    /// The store path to upload.
+    #[bpaf(positional("store-path"))]
+    store_path: PathBuf,
+}
+
+impl Upload {
+    pub async fn handle(self, config: Config, flox: Flox) -> Result<()> {
+        if !config.features.unwrap_or_default().upload {
+            message::plain("🚧 👷 heja, a new command is in construction here, stay tuned!");
+            bail!("'upload' feature is not enabled.");
+        }
+
+        let UploadStorePath { store_path } = self.store_path;
+
+        Self::upload(flox, store_path, self.cache).await
+    }
+
+    #[instrument(name = "upload", skip_all, fields(package))]
```

Comment:
suggestion: we tend to apply the instrumentation the the `handle` method of the individual commands, unless we handle additional subcommands.

Besides, i'm not sure what the `package` field refers to, i think this might be a leftover from copying the publish command?